lib/helpers/wp-cli-db/class-wp-cli-db.php

```diff
@@ -30,40 +31,74 @@ public function early_init() {
 	}
 
 	/**
-	 * Ensure the command is allowed for the current Config.
+	 * Ensure the command or query is allowed for the current Config.
 	 *
-	 * @throws Exception if the command is not allowed.
+	 * @param array $command
+	 * @return WP_Error|null
 	 */
-	public function validate_subcommand( array $command ): void {
+	public function validate_subcommand( array $command ): ?WP_Error {
 		$subcommand = $command[1] ?? '';
 
 		if ( 'query' !== $subcommand ) {
-			throw new Exception( 'ERROR: Only the `wp db query` subcommand is permitted for this site.' );
+			return new \WP_Error( 'db-cli-disallowed-subcmd', 'Only the `wp db query` subcommand is permitted for this site.' );
```

Comment:
I don't think this is correct, there are some other commands that are used  in other parts of the stack like `wp db prefix`, and others are pretty innocuous.  We should come up with an explicit list.

Reply:
@rinatkhaziev What's the best way to go about compiling that list? 🙂